sdk/python/kfp_tekton/compiler/compiler.py

```diff
@@ -40,9 +40,16 @@
 
 def _get_super_condition_template():
 
-  python_string = 'python -c \'import sys\ninput1=str.rstrip(sys.argv[1])\ninput2=str.rstrip(sys.argv[2])\n' \
-    + 'try:\n  input1=int(input1)\n  input2=int(input2)\nexcept:\n  input1=str(input1)\n' \
-    + 'sys.exit(0) if (input1 $(params.operator) input2) else sys.exit(1)\' \'$(params.operand1)\' \'$(params.operand2)\''
+  python_string = textwrap.dedent('''\
```

Comment:
Thanks @drewbutlerbb4  this looks much nicer :-)  ...but I would call it `python_script` as opposed to `python_string`

Reply:
it looks like this new script created an error, here's the log
```
+ python -c import sys
input1=str.rstrip(sys.argv[1])
input2=str.rstrip(sys.argv[2])
try:
  input1=int(input1)
  input2=int(input2)
except:
  input1=str(input1)
sys.exit(0) if (input1 == input2) else sys.exit(1) tails
 heads
```

Reply:
maybe is better to rewrite the static template to have it working first. Then we can port the string from the template into this code.

Reply:
I'm not seeing that error on my end, would you mind giving more information @Tomcli? This issue is at compile time or runtime?

Reply:
it happened during runtime, I'm using the same unit test example in this PR. My Tekton version is 0.12.0

Reply:
Oh sorry I think Tekton intended to show errors. The pipeline status said "completed" instead of "Succeeded", so it was breaking my API code.

Reply:
We probably need to update the API and UI code to catch these new status.

Reply:
Sure, I can update this for the UI. 'completed' is not a valid status for kfp there don't seem to be any direct equivalents but 'Terminated' seems to be the closest.